tests/DbRetryProxyTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractor\Tests;
+
+use Keboola\DbExtractor\DbRetryProxy;
+use Keboola\DbExtractorLogger\Logger;
+use Monolog\Handler\TestHandler;
+use PHPUnit\Framework\TestCase;
+
+class DbRetryProxyTest extends TestCase
+{
+    /** @var TestHandler */
+    private $testHandler;
+
+    /** @var Logger */
+    private $logger;
+
+    public function setUp(): void
+    {
+        $this->testHandler = new TestHandler();
+        $this->logger = new Logger('dbRetryProxyTest');
+        $this->logger->pushHandler($this->testHandler);
+    }
+
+    public function testMaxRetries(): void
```

Comment:
#protip if you named the test "what I expect to happen" you'd noticed that there should be more tests. 

`testWillRetryAndSucceedIfMaxRetriesNotReached`
`testWillThrowAfterDefaultNumberOfRetriesReached`
`testCanOverrideDefaultNumberOfRetries`
`testCanOverrideDefaultTimeout`
`testMaxRetriesIsIgnoredWhenCustomRetryPolicyIsUsed`
... ;)